OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -943,6 +942,53 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 						node.Key = node.Key.Replace("ProvidesCustomPrerequisite", "ProvidesPrerequisite");
 				}
 
+				if (engineVersion < 20150516)
+				{
+					// Replace numbers with strings for DeathSounds.DeathType
+					if (node.Key.StartsWith("DeathSounds"))
+					{
+						var deathTypes = node.Value.Nodes.FirstOrDefault(x => x.Key == "DeathTypes");
+						if (deathTypes != null)
+						{
+							var types = FieldLoader.GetValue<string[]>("DeathTypes", deathTypes.Value.Value);
+							deathTypes.Value.Value = string.Join(", ", types.Select(type => "DeathType" + type));
+
+							var mod = Game.ModData.Manifest.Mod.Id;
+							if (mod == "cnc" || mod == "ra")
+							{
+								deathTypes.Value.Value = deathTypes.Value.Value.Replace("DeathType1", "DefaultDeath");
+								deathTypes.Value.Value = deathTypes.Value.Value.Replace("DeathType2", "Bullet");
```

Comment:
Can we make these `BulletDeath`, `FireDeath` etc?  This would avoid problems down the track where we may want warheads that share a death type but do other different things.